src/change_stream.ts

```diff
@@ -310,7 +310,7 @@ export class ChangeStream<TSchema extends Document = Document> extends TypedEven
   }
 
   /** Check if there is any document still available in the Change Stream */
-  hasNext(callback?: Callback): Promise<void> | void {
+  hasNext(callback?: Callback): Promise<boolean> | void {
```

Comment:
```suggestion
  hasNext(): Promise<boolean>;
  hasNext(callback: Callback<boolean>): void;
  hasNext(callback?: Callback): Promise<boolean> | void {
```

I think this would increase usability even more, dividing up the return type depending on whether or not a callback was provided.